src/elements/e-video.vue

```diff
@@ -0,0 +1,201 @@
+      - vimeo not able to disable controls
+    size:
+      - width & height can either be responsive or hard coded
+      - youtube standard -> width="560", height="315" -->
+  <div :class="b()">
+    <!-- <div style="padding: 56.25% 0 0 0; position: relative"> -->
+    <!-- style="position: absolute; top: 0; left: 0; width: 100%; height: 100%" -->
+    <iframe
+      :class="b('iframe')"
+      :src="mapSrc"
+      :allowfullscreen="allowfullscreen"
+      :frameborder="frameborder.toString()"
+      :width="width.toString()"
+      :height="height.toString()"
+    ></iframe>
+    <!-- </div> -->
+  </div>
+</template>
+
+<script lang="ts">
+  import { PropType, defineComponent } from 'vue';
+
+  const videoUrlSources = {
+    youtube: 'https://www.youtube.com',
+    vimeo: 'https://vimeo.com/',
+  };
+
+  const playerUrlSources = {
+    youtube: 'https://www.youtube.com/embed/',
+    vimeo: 'https://player.vimeo.com/video/',
+  };
+
+  export default defineComponent({
+    name: 'e-video',
+    status: 0, // TODO: remove when component was prepared for current project.
+
+    // components: {},
+
+    props: {
+      /**
+       * Allows the user to watch the video in fullscreen
+       */
+      allowfullscreen: {
+        type: Boolean,
+        default: false,
+      },
+
+      /**
+       * Specifies the source of the video
+       *
+       * We can decide between youtube and vimeo
+       */
+      source: {
+        type: String as PropType<'youtube' | 'vimeo'>,
+        required: true,
+        validator(value: string) {
+          return Object.keys(playerUrlSources).includes(value);
+        },
+      },
+
+      /**
+       * Allows to add/remoe the border around the player
+       *
+       * When set to 0 the border is completely removed
+       *
+       * Default from iframe is 2
+       */
+      frameborder: {
+        type: [String, Number],
+        default: 2,
+      },
+
+      /**
+       * Accepts a valid url from a youtube or vimeo video
+       */
+      videoUrl: {
+        type: String,
+        default: null,
+      // validate against different sources
+      // validator: (value: string) => playerSources[this.source].includes(value)
+      // validator: (value: string) => {
+      //   return Object.values(videoUrlSources).includes(value);
+      // }
+      },
+
+      /**
+       * Accepts an ID from a youtube or vimeo video
+       */
+      videoId: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Allows to set the video player width.
+       */
+      width: {
+        type: [String, Number],
+        default: 560,
+      },
+
+      /**
+       * Allows to set the video player height.
+       */
+      height: {
+        type: [String, Number],
+        default: 315,
+      },
+
+    /**
+     * Defines the players size to be responsive 100% width to the parent container
+     */
+    // responsive: {
+    //   type: Boolean,
+    //   default: null
+    // }
```

Comment:
Either implement the feature or remove the code. I think it would be a nice feature to have. Another way could be to make this the default behavior and remove the defaults for width and height.

Reply:
I do think having the possibility is a plus. So I added the property and functionality to this PR.
For now it still defaults to height and weight.

Let me know if this is fine or it should fallback to `responsive="true"` if neither `responsive` or `height`+ `width` are specified.